src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size = 0;
+    private Node<K,V>[] table;
 
     @Override
     public void put(K key, V value) {
+        if (this.table == null || (this.size + 1) > this.table.length * LOAD_FACTOR) {
```

Comment:
```suggestion
        if (size >= table.length * LOAD_FACTOR) {
```
initialize table in constructor and get rid of null checks everywhere